micro_dl/utils/masks.py

```diff
@@ -1,55 +1,67 @@
 import numpy as np
 import scipy.ndimage as ndimage
-import cv2
-import skimage
+from skimage.filters import threshold_otsu, gaussian, laplace
 from scipy.ndimage import binary_fill_holes
-from skimage.filters import threshold_otsu
-from skimage.feature import peak_local_max
-from skimage.morphology import disk, ball, binary_opening, binary_erosion
-from skimage.segmentation import watershed
+from skimage.morphology import (
+    ball,
+    binary_opening,
+    remove_small_objects,
+    binary_dilation,
+)
 from micro_dl.utils.image_utils import im_adjust
 
 
-def create_otsu_mask(input_image, kernel_size=3):
-    """Create a binary mask using morphological operations
+def create_otsu_mask(input_image, sigma=0.6):
 
-    :param np.array input_image: generate masks from this image
-    :param int str_elem_size: size of the structuring element. typically 3, 5
-    :return: mask of input_image, np.array
+    """Create a binary mask using morphological operations
+    :param np.array input_image: generate masks from this 3D image
+    :param float sigma: Gaussian blur standard deviation, increase in value increases blur
+    :return: volume mask of input_image, 3D np.array
     """
 
-    input_image = im_adjust(
-        cv2.GaussianBlur(input_image, (kernel_size, kernel_size), 0)
+    input_image_blur = gaussian(input_image, sigma=sigma)
+
+    input_sz = input_image.shape
+    mid_slice_id = (np.ceil(input_sz[0]/2)).astype(int)
```

Comment:
Consider 5 slices, `(np.ceil(5/2)).astype(int)==3` where you would want index 2. Maybe just use [this](https://github.com/mehta-lab/recOrder/blob/6742be9bf99e2d05d00213d722798dce2e8602f5/recOrder/compute/reconstructions.py#L169-L170) instead?

Reply:
Thank you! I have fixed this.